src/main/java/com/marklogic/spark/reader/optic/OpticReadContext.java

```diff
@@ -130,20 +172,55 @@ private PlanBuilder.Plan buildPlanForBucket(RowManager rowManager, PlanAnalysis.
             .bindParam("ML_LOWER_BOUND", bucket.lowerBound)
             .bindParam("ML_UPPER_BOUND", bucket.upperBound);
 
+        // Add user-defined parameters from options
+        for(Map.Entry<String, String> entry : this.userDefinedParams.entrySet()) {
+            if (logger.isDebugEnabled()) {
+                logger.debug("Adding user param: {}", entry);
+            }
+            plan = plan.bindParam(entry.getKey(), entry.getValue());
+        }
+
         if (opticFilters != null) {
             for (OpticFilter opticFilter : opticFilters) {
                 plan = opticFilter.bindFilterValue(plan);
             }
         }
 
+        for (Map.Entry<String, String> entry : columnParamValues.entrySet()) {
```

Comment:
This should go into a new private method with a name that describes what it's doing - e.g. `bindUserParamsToPlan`. 

And I think definitely get rid of "column" in the variable names. I think `userParams` is a good term to use across the board - i.e. "op.param calls that are defined by a user in the Optic query, and which the user is responsible for providing a value for at runtime (with support for a default value as well)". The term "column" doesn't add any value.